app/tests/app/models/variable.service.test.js

```diff
@@ -0,0 +1,368 @@
+"use strict";
+
+fdescribe("Service: Variable ", function() {
+  beforeEach(module("app.config"));
+  beforeEach(module("chuvApp.components.criteria"));
+
+  var Variable, $resource, $http, $cacheFactory, $q, $httpBackend;
+  var makeVariableService;
+
+  beforeEach(function() {
+    jasmine.addMatchers({
+      toEqualData: function() {
+        return {
+          compare: function(actual, expected) {
+            return { pass: angular.equals(actual, expected) };
+          }
+        };
+      }
+    });
+  });
+
+  beforeEach(
+    inject(function(
+      _$resource_,
+      backendUrl,
+      _$http_,
+      _$cacheFactory_,
+      _$q_,
+      _Variable_,
+      _$httpBackend_
+    ) {
+      $resource = _$resource_;
+      $http = _$http_;
+      $cacheFactory = _$cacheFactory_;
+      $q = _$q_;
+      Variable = _Variable_;
+      $httpBackend = _$httpBackend_;
+    })
+  );
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it("Should return the whole hierarchical data when hierarchy() is called", function(
+    done
+  ) {
+    $httpBackend
```

Comment:
here note the difference with when("GET".  expectGET wants to be called once and only once, so our test is more precise, and we can detect wild multiple requests etc